BUILD/capture_warnings.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+TMP_FILE=/tmp/compile.warning
+WARN_PATH=$1
+shift
+WARN_FILE="$WARN_PATH/compile.warnings"
+
+set -o pipefail
+rm -f $TMP_FILE
+$@ 2> $TMP_FILE
```

Comment:
1. Use this technique to catch `STDERR` to variable:
    ```
    exec 3>&1
    x=$((echo err >&2; echo out) 2>&1 1>&3)
    ```

2. Wrong expansion of `$@`. Use `"$@" instead`.